include/zephyr/net/ethernet.h

```diff
@@ -465,6 +467,26 @@ struct ethernet_txtime_param {
 	bool enable_txtime;
 };
 
+/** Protocols that are supported by checksum offloading */
+enum ethernet_checksum_support {
+	/** Device does not support any L3/L4 checksum offloading */
+	ETHERNET_CHECKSUM_SUPPORT_NONE			= NET_IF_CHECKSUM_NONE_BIT,
+	/** Device supports checksum offloading for the IPv4 header */
+	ETHERNET_CHECKSUM_SUPPORT_IPV4_HEADER		= NET_IF_CHECKSUM_IPV4_HEADER_BIT,
+	/** Device supports checksum offloading for ICMPv4 payload (implies IPv4 header) */
+	ETHERNET_CHECKSUM_SUPPORT_IPV4_ICMP		= NET_IF_CHECKSUM_IPV4_ICMP_BIT,
+	/* Space for future protocols and restrictions for IPV4 */
```

Comment:
This comment doesn't really fit here anymore, should probably be moved to the bits.

Reply:
Done.